From 5db16d4bc394f06fd0e2a5e051f003d1b5cb15ae Mon Sep 17 00:00:00 2001
From: Olli-Antti Kivilahti <olli-antti.kivilahti@jns.fi>
Date: Wed, 4 Jan 2017 12:29:10 +0200
Subject: [PATCH 37/58] KD-116 Juveline/Juvenile material doesn't incur fines
 for any itemtype

Added hacky TPVN to exclusion list via a hacky regexp
---
 C4/Overdues.pm | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/C4/Overdues.pm b/C4/Overdues.pm
index 40f726a..b85d256 100644
--- a/C4/Overdues.pm
+++ b/C4/Overdues.pm
@@ -257,8 +257,10 @@ sub CalcFine {
     #HACKMAN HERE! Childrens and young adults material from permanent_location LAP, LAK, LVA, NUO, NUA is not fined,
     #  unless it is of itemtypes VI, DV, DR, KP, CR, BR
     my $itemRow = GetItem( undef, $item->{barcode}, undef );
+
     my $pl = $itemRow->{permanent_location};
-    if ( ($pl eq 'KUV' || $pl eq 'LAP' || $pl eq 'LAK' || $pl eq 'LVA' || $pl eq 'NUO' || $pl eq 'NUA' || $pl eq 'NUV' || $pl eq 'OHE') ) {
+    my $ic = $itemRow->{itemcallnumber} || '';
+    if ( ($ic =~ /TPVN/ || $pl eq 'KUV' || $pl eq 'LAP' || $pl eq 'LAK' || $pl eq 'LVA' || $pl eq 'NUO' || $pl eq 'NUA' || $pl eq 'NUV' || $pl eq 'OHE') ) {
             $data->{fine} = 0;
     }
 
-- 
1.9.1

